src/Datasource/EntityTrait.php

```diff
@@ -418,24 +418,26 @@ public function visibleProperties()
      * This method will recursively transform entities assigned to properties
      * into arrays as well.
      *
+     * @param bool $includeHiddenProperties Include hidden properties in array
      * @return array
      */
-    public function toArray()
+    public function toArray($includeHiddenProperties = false)
     {
+        $properties = $includeHiddenProperties ? array_merge($this->visibleProperties(), $this->hiddenProperties()): $this->visibleProperties();
```

Comment:
Could this be a + operator? This is usually way faster for string key based arrays.